test/bdd/did_resolver_sidetree_node_steps.go

```diff
@@ -0,0 +1,183 @@
+		if err != nil {
+			return err
+		}
+		msg = strings.Replace(msg, "#didDocumentHash", documentHash, -1)
+	}
+	logger.Infof("check success resp %s contain %s", string(d.resp.payload), msg)
+	if !strings.Contains(string(d.resp.payload), msg) {
+		return fmt.Errorf("success resp %s doesn't contain %s", d.resp.payload, msg)
+	}
+	return nil
+}
+
+func (d *DIDResolverSideTreeNodeSteps) wait(seconds int) error {
+	logger.Infof("Waiting [%d] seconds\n", seconds)
+	time.Sleep(time.Duration(seconds) * time.Second)
+	return nil
+}
+
+func newCreateRequest(didDocumentPath string) *models.Request {
+	operation := models.OperationTypeCreate
+	alg := "ES256K"
+	kid := "#key1"
+	payload := encodeDidDocument(didDocumentPath)
+	signature := "mAJp4ZHwY5UMA05OEKvoZreRo0XrYe77s3RLyGKArG85IoBULs4cLDBtdpOToCtSZhPvCC2xOUXMGyGXDmmEHg"
+	return request(alg, kid, payload, signature, operation)
+}
+
+func request(alg, kid, payload, signature string, operation models.OperationType) *models.Request {
+	header := &models.Header{
+		Alg:       swag.String(alg),
+		Kid:       swag.String(kid),
+		Operation: operation,
+	}
+	req := &models.Request{
+		Header:    header,
+		Payload:   swag.String(payload),
+		Signature: swag.String(signature)}
+	return req
+}
+
+func encodeDidDocument(didDocumentPath string) string {
+	r, _ := os.Open(didDocumentPath)
+	data, _ := ioutil.ReadAll(r)
+	doc, _ := document.FromBytes(data)
+	// add new key to make the document unique
+	doc["unique"] = generateUUID()
+	bytes, _ := doc.Bytes()
+	return base64.URLEncoding.EncodeToString(bytes)
+}
+
+// generateUUID returns a UUID based on RFC 4122
+func generateUUID() string {
+	id := GenerateBytesUUID()
+	return fmt.Sprintf("%x-%x-%x-%x-%x", id[0:4], id[4:6], id[6:8], id[8:10], id[10:])
+}
+
+type httpRespone struct {
+	payload  []byte
+	errorMsg string
+}
+
+// SendRequest sends a regular POST request to the sidetree-node
+// - If post request has operation "create" then return sidetree document else no response
+func sendRequest(url string, req *models.Request) (*httpRespone, error) {
+	resp, err := sendHTTPRequest(url, req)
+	if err != nil {
+		return nil, err
+	}
+	return handleHttpResp(resp)
+}
+
+func handleHttpResp(resp *http.Response) (*httpRespone, error) {
+	if status := resp.StatusCode; status != http.StatusOK {
+		r := &models.Error{}
+		err := decode(resp, r)
+		if err != nil {
+			return nil, err
+		}
+		return &httpRespone{errorMsg: swag.StringValue(r.Message)}, nil
+	}
+
+	r := &models.Response{}
+	err := decode(resp, r)
+	if err != nil {
+		return nil, err
+	}
+	payload, err := r.MarshalBinary()
+	if err != nil {
+		return nil, err
+	}
+	return &httpRespone{payload: payload}, nil
+}
+
+func sendHTTPRequest(url string, req *models.Request) (*http.Response, error) {
+	client := &http.Client{}
+	b, err := req.MarshalBinary()
+	if err != nil {
+		return nil, err
+	}
+
+	httpReq, err := http.NewRequest("POST", url, bytes.NewReader(b))
```

Comment:
"POST" is a constant in the http package.

Reply:
done